### PR TITLE
Cleanup new features API with env support

### PR DESCRIPTION
**Pull request by ivarconr**

### CODE REVIEW

File: src/lib/db/feature-strategy-store.ts [hunk 1]
```diff
@@ -301,99 +270,65 @@ class FeatureStrategiesStore implements IFeatureStrategiesStore {
...
                 'feature_environments.feature_name',
                 'features.name',
             )
             .fullOuterJoin(
-                'feature_strategies',
-                'feature_strategies.feature_name',
-                'features.name',
+                'environments',
+                'feature_environments.environment',
+                'environments.name',
```
Comment (chriswk) [inline src/lib/db/feature-strategy-store.ts:306]: Nice catch

File: src/lib/db/feature-toggle-client-store.ts [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+    description: string;
+    type: string;
+    stale: boolean;
+    variants: string;
+    project: string;
+    last_seen_at?: Date;
+    created_at?: Date;
+}
+
+export default class FeatureToggleClientStore
```
Comment (chriswk) [inline src/lib/db/feature-toggle-client-store.ts:24]: Good to separate out the client specific queries :)

File: src/lib/db/project-store.ts [hunk 3]
```diff
@@ -159,7 +155,7 @@ class ProjectStore implements IProjectStore {
             .where({
                 project_id: id,
             })
-            .returning('environment_name');
+            .pluck('environment_name');
```
Comment (chriswk) [inline src/lib/db/project-store.ts:158]: TIL we have pluck :+1:

File: src/lib/routes/admin-api/environments-controller.ts [hunk 4]
```diff
@@ -32,62 +31,42 @@ export class EnvironmentsController extends Controller {
...
 
     async getAll(req: Request, res: Response): Promise<void> {
-        try {
-            const environments = await this.service.getAll();
-            res.status(200).json({ version: 1, environments });
-        } catch (e) {
-            handleErrors(res, this.logger, e);
-        }
+        const environments = await this.service.getAll();
+        res.status(200).json({ version: 1, environments });
```
Comment (chriswk) [inline src/lib/routes/admin-api/environments-controller.ts:35]: Love the new controller cleanup. So much easier to read and write :+1:

File: src/lib/routes/client-api/feature.test.ts [hunk 5]
```diff
@@ -130,10 +130,10 @@ test('fetch single feature', async () => {
...
 test('support name prefix', async () => {
     expect.assertions(2);
-    await featureStrategiesStore.createFeature({ name: 'a_test1' });
-    await featureStrategiesStore.createFeature({ name: 'a_test2' });
-    await featureStrategiesStore.createFeature({ name: 'b_test1' });
-    await featureStrategiesStore.createFeature({ name: 'b_test2' });
+    await featureToggleClientStore.createFeature({ name: 'a_test1' });
+    await featureToggleClientStore.createFeature({ name: 'a_test2' });
+    await featureToggleClientStore.createFeature({ name: 'b_test1' });
+    await featureToggleClientStore.createFeature({ name: 'b_test2' });
```
Comment (chriswk) [inline src/lib/routes/client-api/feature.test.ts:136]: I'm confused, why is the clientStore creating features? I thought it was meant only for the /api/client endpoints?

Comment (chriswk) [reply src/lib/routes/client-api/feature.test.ts:136]: Wouldn't createFeature be more suited for a featureToggleAdminStore?

File: src/test/e2e/api/admin/bootstrap.test.ts [hunk 6]
```diff
@@ -8,7 +8,7 @@ let db;
 const email = 'user@getunleash.io';
 
 beforeAll(async () => {
-    db = await dbInit('user_api_serial', getLogger);
+    db = await dbInit('ui_bootstrap_serial', getLogger);
```
Comment (chriswk) [inline src/test/e2e/api/admin/bootstrap.test.ts:11]: the copy pasting curse :)

**Review by chriswk — APPROVED** [review]
This looks good to me. Besides my confusion with the featureStores I think we're good to go

File: src/lib/routes/client-api/feature.test.ts (hunk 5 above)
Comment (ivarconr) [reply src/lib/routes/client-api/feature.test.ts:136]: I totally agree, I am even unsure if this actually works, or there might be a typo in its name.

File: src/lib/db/project-store.ts (hunk 3 above)
Comment (ivarconr) [reply src/lib/db/project-store.ts:158]: was new to me too. Had to find a fix as this one was not returning what we wanted and I start exploring the docs and discovered it by chance. I think it was added to knex recently.

File: src/lib/routes/admin-api/environments-controller.ts (hunk 4 above)
Comment (ivarconr) [reply src/lib/routes/admin-api/environments-controller.ts:35]: jup, have been thinking abut this for a while, but was a bit tricky to get the typing's correct with express and typescript.

File: src/lib/routes/client-api/feature.test.ts (hunk 5 above)
Comment (ivarconr) [reply src/lib/routes/client-api/feature.test.ts:136]: So this is where our fake-stores got limitations..... one fake store cannot insert data to another fake store. Suggestions?

Comment (ivarconr) [reply src/lib/routes/client-api/feature.test.ts:136]: I will rename it to _createFeature for now and add a comment.

**Pull request merged**